x-pack/plugins/stack_alerts/server/alert_types/es_query/alert_type_params.ts

```diff
@@ -66,8 +94,8 @@ function validateParams(anyParams: unknown): string | undefined {
   }
 }
 
-export function validateComparator(comparator: string): string | undefined {
-  if (ComparatorFnNames.has(comparator)) return;
+function validateComparator(comparator: string): string | undefined {
+  if (ComparatorFnNames.has(comparator as Comparator)) return;
```

Comment:
why the cast?

Reply:
We have to improve `@kbn/config-schema` package and introduce generic `schema.string<T>(...)` function. `T` in our case will be `Comparator`. Then we will be able to remove that cast.

Reply:
Issue created #129184